helpers/helpers-spec_service.bal

```diff
@@ -0,0 +1,190 @@
+    }
+
+    resource function post 'base64/decode/[string value]() returns Base64_responseOk|Error_responseBadRequest {
+        // Validate incoming string 
+        log:printDebug("Incoming text: " + value);
+
+        if (value.length() > MAX_BASE64_STRING_SIZE) {
+            Error_responseBadRequest response = {body: {"message": "String is too large. Sorry.", "code": "err_002"}};
+            return response;
+        }
+        string:RegExp pattern = re `^[0-9a-zA-Z=]+$`;
+
+        if ((value.matches(pattern)) is false) {
+            Error_responseBadRequest response = {body: {"message": "Invalid characters. Sorry.", "code": "err_003"}};
+            return response;
+        }
+        string|error decodedValue = mime:base64Decode(value).ensureType(string);
+
+        if (decodedValue is string) {
+            Base64_responseOk response = {body: {"value": decodedValue}};
+            log:printDebug("Decoded Value OK  " + decodedValue);
+            return response;
+        } else {
+            Error_responseBadRequest response = {body: {"message": "unable to decode", "code": "err_004"}};
+            log:printDebug("Error decoding text: " + decodedValue.toString());
+            return response;
+        }
+    }
+
+    resource function post 'base64/encode/[string value]() returns Base64_responseOk|Error_responseBadRequest {
+        log:printDebug("Inbound Value  " + value);
+
+        // Validate incoming string 
+        if (value.length() > MAX_BASE64_STRING_SIZE) {
+            Error_responseBadRequest response = {body: {"message": "String is too large. Sorry.", "code": "err_002"}};
+            return response;
+        }
+        string:RegExp pattern = re `^[0-9a-zA-Z\\s!$-_]+$`;
+
+        if (pattern.isFullMatch(value) is false) {
+            Error_responseBadRequest response = {body: {"message": "Invalid characters. Sorry.", "code": "err_003"}};
+            return response;
+        }
+
+        string|error encodedValue = mime:base64Encode(value).ensureType(string);
+
+        if (encodedValue is string) {
+            Base64_responseOk response = {body: {"value": encodedValue}};
+            log:printDebug("Encoded Value: " + encodedValue);
+            return response;
+        }
+        else {
+            Error_responseBadRequest response = {body: {"message": "unable to encode", "code": "err_006"}};
+            log:printDebug("Error encoding text:  " + encodedValue.toString());
+            return response;
+        }
+    }
+
+    resource function post ai/spelling(@http:Payload ai_spelling_payload data) returns error|ai_spelling_responseOk|Error_responseBadRequest {
+        http:RetryConfig retryConfig = {
+            interval: 5, // Initial retry interval in seconds.
+            count: 3, // Number of retry attempts before stopping.
+            backOffFactor: 2.0 // Multiplier of the retry interval.
+        };
+
+        final chat:Client openAIChat = check new ({auth: {token: OPENAI_KEY}, retryConfig});
```

Comment:
Can be created at module-level, right? Or we'll be creating a new client for each request to this resource?